### ISSUE TITLE
Set c-basic-offset cannot change c-c++ file indentation(ver 0.3000)

### ISSUE THREAD

**Issue by Voleking** [syl20bnr/spacemacs#10051]
#### Description :octocat:
After upgrade to Spacemacs 0.3000, Set c-basic-offset cannot change c-c++ file indentation.

#### Reproduction guide :beetle:
- Start Emacs
- open a c/cpp file

*Observed behaviour:* :eyes: :broken_heart:
indent with "2" space

*Expected behaviour:* :heart: :smile:
indent with "4" space

#### System Info :computer:
- OS: darwin
- Emacs: 25.3.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. b4d663dc)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm osx nlinum
      (chinese :variables chinese-enable-fcitx t)
      dash git github pandoc
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode c-c++-enable-clang-support nil)
      emacs-lisp
      (markdown :variables markdown-command "pandoc -t html5 -f markdown -s -S --mathjax --highlight-style=pygments --toc --toc-depth 3 --template github.html5 --css html/css/github.css" markdown-live-preview-engine 'pandoc)
      (org :variables org-agenda-files
           '("~/Documents/Org")
           org-directory '"~/Documents/Org" org-projectile-file "TODOs.org" org-want-todo-bindings t)
      (python :variables python-enable-yapf-format-on-save t)
      (auto-completion :variables auto-completion-enable-help-tooltip 'manual auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t :disabled-for org markdown)
      (shell :variables shell-default-shell 'multi-term shell-default-height 80 shell-default-position 'bottom)
      (syntax-checking :variables flycheck-delayed-errors-timeout 500)
      (wakatime :variables wakatime-api-key "2ccf4cb6-4369-40b5-9165-718666b8bb32" wakatime-cli-path "/usr/local/bin/wakatime")
      cfg)
```
- System configuration features: NOTIFY ACL LIBXML2 ZLIB TOOLKIT_SCROLL_BARS
- config:
```elisp
(setq c-basic-offset 4)
(setq tab-width 4)
(c-set-offset 'case-label 4)
```
I aso try `setq-default`.

#### Describe Variable in c-cpp file:
I got this:
```
c-basic-offset is a variable defined in ‘cc-vars.el’.
Its value is 2
Original value was set-from-style
Local in buffer C.cpp; global value is 4
```

**Comment by Voleking** [comment]
I found `google-c-style` in `c-c++-layer`

**Comment by AngelTerrones** [comment]
@Voleking Hi. How did you fix the indentation?

**Comment by Voleking** [comment]
@AngelTerrones add `google-c-style` to `dotspacemacs-excluded-packages` in `init.el` or change `(c-basic-offset . 2)` to `(c-basic-offset . 4)` in `google-c-style.el` will fix the indentation.

**Comment by AngelTerrones** [comment]
Ok, gonna try. Thx.

**Comment by aurabindo** [comment]
This didnt fix my issue. Besides, how the change in google-style-c fix my offset when I am trying to set a different one ? Is google-style-c the default ?

**Comment by AngelTerrones** [comment]
@aurabindo It seems. I added `google-c-style` to `dotspacemacs-excluded-packages`. Worked for me.

**Comment by Voleking** [comment]
@aurabindo Yech, see the code [here](https://github.com/syl20bnr/spacemacs/blob/b3ddc962ecad0bde91a75f609cd505021ced17df/layers/%2Blang/c-c%2B%2B/packages.el#L251). Since it's `add-hook` so `(setq c-basic-offset 4)` doesn't work anymore. If you don't want to use `google-style-c` as default, maybe add `google-c-style` to `dotspacemacs-excluded-packages` is the simplest way.

**Comment by aurabindo** [comment]
@Voleking Adding google-c-style to excluded package fixes the problem as you said. Thank you.

**Comment by myme** [comment]
This should now have been resolved in develop now with #10190.

**Comment by sunlin7** [comment]
I use these slice of code to modify the `c-basic-indent` without exclude the google-c-style,
(eval-after-load 'google-c-style
```elisp
(eval-after-load 'google-c-style
    (dolist (v google-c-style)
      (when (and (listp v) (eq (car v) 'c-basic-offset)) 
        (setcdr v 4))))
```

**Comment by c02y** [comment]
Excluding `google-c-style` package is not working for me, it is already excluded in my init.el. I spent multiple hours diagnosing this problem, I got my own setup for indentation (c-basic-offset, tab-width, etc.) for c/c++ code, but if I enable both c-c++ and lsp layers, the indent function like `indent-region` will not use my indention setup, no matter what lsp-backend(clangd, ccls, cquery) I'm using. 

So I diagnose code of my init.el, line by line, code of c-c++ layer of spacemacs, and then found out this is the problem of lsp function(https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Blang/c-c%2B%2B/funcs.el#L89) which is from lsp-mode package, and then I diagnose that code of lap-mode, and search the issue page of lsp-mode, found the solution(https://github.com/emacs-lsp/lsp-mode/issues/398), if any one has this same problem, just put 
~~~
     (lsp :variables lsp-enable-indentation nil)
~~~
into your init.el. the function like indent-region-of-buffer will work OK now.

I think this can be put into the README.org file.
